clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,369 @@
+from cmath import nan
+from os import PathLike
+from typing import BinaryIO, Iterable, List, Tuple, Union
+
+from pandas import DataFrame, notna
+
+
+def find_clinical_data(
+    clinical_data_directory: PathLike,
+) -> DataFrame:
+    from pathlib import Path
+
+    import pandas as pd
+
+    from clinica.utils.stream import cprint
+
+    # Read the xls
+    try:
+        image_data_file = list(Path(clinical_data_directory).glob("*.tsv"))
+    except StopIteration:
+        raise FileNotFoundError("Clinical data file not found.")
+
+    # Assign the dfs
+    df_list = []
+    for i in range(0, len(image_data_file)):
+        df_list.append(pd.read_csv(str(image_data_file[i]), sep="\t"))
+        df_list[i] = df_list[i].set_index(df_list[i].axes[1][0])
+        if df_list[i].index.name == "eid":
+            df_clinical = df_list[i]
+
+    # Warn if a dataframe is missing
+    if "df_clinical" in locals():
+        cprint(msg="All clinical data have been found", lvl="info")
+    else:
+        raise FileNotFoundError("Clinical data not found or incomplete")
+
+    return df_clinical
+
+
+def read_clinical_data(
```

Comment:
Is there a reason you created this function? It looks like it is not doing anything more than `find_clinical_data`.

Reply:
Same as others, it comes from the structure of the original converter the code is inspired from. you are right i could do without.